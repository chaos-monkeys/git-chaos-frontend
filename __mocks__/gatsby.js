/* eslint-disable @typescript-eslint/no-unused-vars */
import React from "react";

const gatsby = jest.requireActual("gatsby");

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  // these props are invalid for an `a` tag
  Link: jest
    .fn()
    .mockImplementation(
      ({
        activeClassName,
        activeStyle,
        getProps,
        innerRef,
        partiallyActive,
        ref,
        replace,
        to,
        ...rest
      }) =>
        /* istanbul ignore next */
        React.createElement("a", {
          ...rest,
          href: to,
        }),
    ),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn(),
};
