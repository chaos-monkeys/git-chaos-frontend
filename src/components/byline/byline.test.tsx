import React from "react";
import { render, getByTestId } from "@testing-library/react";
import renderer from "react-test-renderer";
import * as Gatsby from "gatsby";
import Byline from "./byline";

describe("Byline", () => {
  const email = "feedback@chaosmonkeys.dev";

  beforeEach(() => {
    const data = {
      site: {
        siteMetadata: {
          emails: {
            feedback: email,
          },
        },
      },
    };

    const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery");
    useStaticQuery.mockImplementationOnce(() => data);
  });

  it("matches snapshot", () => {
    // unable to do 'toMatchInlineSnapshot' due to eslint removing white space, and creating a diff :(
    expect(renderer.create(<Byline />).toJSON()).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { container } = render(<Byline />);
    expect(getByTestId(container, "byline")).toBeInTheDocument();
    const link = getByTestId(container, "byline-link");
    expect(link.getAttribute("href")).toBe(`mailto:${email}`);
  });
});
