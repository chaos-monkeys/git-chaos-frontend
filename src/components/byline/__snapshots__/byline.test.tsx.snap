// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Byline matches snapshot 1`] = `
<div
  data-testid="byline"
>
  <p>
    Chaos Monkeys is a built with
     
    <span
      aria-label="love"
      role="img"
    >
      ❤️
    </span>
     
    by Ollie & Rob.
  </p>
  <p>
    We’re building kickass tools to help you deliver better. We want you to love what we build as much as we do -
     
    <a
      data-testid="byline-link"
      href="mailto:feedback@chaosmonkeys.dev"
    >
      let us know how we are doing
    </a>
  </p>
</div>
`;
